operator/pkg/lbipam/lbipam.go

```diff
@@ -1089,8 +1089,13 @@ func (ipam *LBIPAM) handleNewPool(ctx context.Context, pool *cilium_api_v2alpha1
 		if fromCidr && pool.Spec.AllowFirstLastIPs != "yes" {
 			// TODO: in 1.16 switch from default no to default yes.
 			from, to := lbRange.alloc.Range()
-			lbRange.alloc.Alloc(from, true)
-			lbRange.alloc.Alloc(to, true)
+
+			// If the first and last IPs are the same or adjacent, we would reserve the entire range.
+			// Only reserve first and last IPs for ranges /30 or /126 and larger.
+			if !(from.Compare(to) == 0 || from.Next().Compare(to) == 0) {
```

Comment:
I'm wondering, if this is meant to be exclusive at both ends by default, should we just consider using `allowFirstLastIPs`when the ranges are /32?

Reply:
I don't understand the question sorry. This piece of code is to always allow first and last IP for /32 and /31 ranges because reserving the full range is always nonsensical